src/main/java/nextstep/subway/line/domain/Section.java

```diff
@@ -86,4 +67,36 @@ public boolean isMatchDownStation(final Station targetStation) {
     public void addToPath(final Consumer<Section> consumer) {
         consumer.accept(this);
     }
+
+    public boolean findMatchingSection(final List<Station> stations) {
+        if (stations.stream().noneMatch(this::isMatchUpStation)) {
+            return false;
+        }
+
+        return stations.stream().anyMatch(this::isMatchDownStation);
```

Comment:
요거는 한줄로 해도 똑같은 결과가 나오지 않을까요?

boolean | allMatch(Predicate<? super T> predicate) | Returns whether all elements of this stream match the provided predicate.
-- | -- | --
boolean | anyMatch(Predicate<? super T> predicate) | Returns whether any elements of this stream match the provided predicate.
boolean | noneMatch(Predicate<? super T> predicate) | Returns whether no elements of this stream match the provided predicate.

noneMatch는 일치하는게 하나도 없는 경우
anyMatch는 적어도 하나는 일치하는 경우

anyMatch만 사용해도 0개인 경우 -> false, 1개라도 있는 경우 -> true로 나오지 않을까요?

```suggestion
        return stations.stream().anyMatch(this::isMatchDownStation);
```

적다보니 저도 헷갈리는데.. 한번 확인해보시고 코멘트 남겨주세요!

Reply:
아.. 그렇네요.. 제가 이부분은 새벽에 해가지고.. 헷갈린것 같습니다!

하나의 스테이션을 upStation, downStaion을 비교할 필요가 없는건데..
해당 부분은 다시 짜서 반영하겠습니다! ㅠ

지적 감사합니다~